Index: top.inc.php
===================================================================
RCS file: /usr/local/cvsroot/eshop/top.inc.php,v
retrieving revision 1.17.2.12
diff -u -r1.17.2.12 top.inc.php
--- top.inc.php	5 Feb 2009 10:39:23 -0000	1.17.2.12
+++ top.inc.php	13 Apr 2010 14:00:04 -0000
@@ -301,5 +301,20 @@
 	)
 );
 
+define('XC_TIME', time());
+
+define('X_PAYMENT_TRANS_ALREADY_CAPTURED', 1);
+define('X_PAYMENT_TRANS_ALREADY_VOIDED', 2);
+define('X_PAYMENT_TRANS_ALREADY_REFUNDED', 4);
+define('X_PAYMENT_TRANS_ALREADY_ACCEPTED', 8);
+define('X_PAYMENT_TRANS_ALREADY_DECLINED', 16);
+
+define('PAYMENT_NEW_STATUS', 1);
+define('PAYMENT_AUTH_STATUS', 2);
+define('PAYMENT_DECLINED_STATUS', 3);
+define('PAYMENT_CHARGED_STATUS', 4);
+
+define('X_USE_PAYPAL_FLOW', true);
+
 }
 ?>
Index: admin/cc_processing.php
===================================================================
RCS file: /usr/local/cvsroot/eshop/admin/cc_processing.php,v
retrieving revision 1.56.2.15
diff -u -r1.56.2.15 cc_processing.php
--- admin/cc_processing.php	5 Feb 2009 10:39:23 -0000	1.56.2.15
+++ admin/cc_processing.php	13 Apr 2010 14:00:04 -0000
@@ -42,10 +42,22 @@
 
 x_load('backoffice','crypt','tests');
 
+if (!empty($active_modules['XPayments_Connector'])) {
+    func_xpay_func_load();
+}
+
 if ($mode=="add" && !empty($processor)) {
 	require $xcart_dir."/include/safe_mode.php";
 
-	$tmp = func_query_first("SELECT * FROM $sql_tbl[ccprocessors] WHERE processor='".$processor."'");
+	$where_query = '';
+	if (preg_match("/cc_xpc\.php_([0-9]+)/", $processor, $match)) {
+		$pm_id = intval($match[1]);
+		$processor = "cc_xpc.php";
+		$where_query = ' AND param01=\'' . $pm_id . '\'';
+	}
+
+	$tmp = func_query_first("SELECT * FROM $sql_tbl[ccprocessors] WHERE processor='".$processor."'" . $where_query);
+
 	$cc_processor = $tmp["module_name"];
 	if (!zerolen($tmp["c_template"])) {
 		$template = $tmp["c_template"];
@@ -66,6 +78,10 @@
 		'processor_file' => $processor
 	);
 
+	if ($processor == 'cc_xpc.php') {
+		$insert_params['protocol'] = 'https';
+	}
+
 	if ($processor == 'ps_paypal.php') {
 		// Paypal standard
 		$paymentid = func_array2insert('payment_methods', $insert_params);
@@ -86,7 +102,7 @@
 	}
 	else {
 		$paymentid = func_array2insert('payment_methods', $insert_params);
-		db_query("UPDATE $sql_tbl[ccprocessors] SET paymentid='".$paymentid."' WHERE processor='".$processor."'");
+		db_query("UPDATE $sql_tbl[ccprocessors] SET paymentid='".$paymentid."' WHERE processor='".$processor."'" . $where_query);
 	}
 
 	func_header_location("payment_methods.php");
@@ -231,7 +247,15 @@
 			$conf_data[$pkey] = $cc_processing_module;
 			$conf_data[$akey] = func_query_first("SELECT $sql_tbl[ccprocessors].* FROM $sql_tbl[ccprocessors] WHERE $sql_tbl[ccprocessors].processor = '$asearch'");
 			$smarty->assign('conf_data', $conf_data);
+
+		} elseif ($cc_processor == 'cc_xpc.php') {
+
+			$cc_processors = func_query("SELECT * FROM $sql_tbl[ccprocessors] WHERE paymentid>0 AND processor='cc_xpc.php'");
+			$smarty->assign('cc_processors', $cc_processors);
+			$cc_processing_module['module_name'] = 'X-Payments payment methods';
+
 		}
+
 	} elseif ($subscribe == "yes") {
 		db_query("UPDATE $sql_tbl[config] SET value='$cc_processor' WHERE name='active_subscriptions_processor'");
 		$config["active_subscriptions_processor"] = $cc_processor;
Index: admin/configuration.php
===================================================================
RCS file: /usr/local/cvsroot/eshop/admin/configuration.php,v
retrieving revision 1.81.2.31
diff -u -r1.81.2.31 configuration.php
--- admin/configuration.php	5 Feb 2009 10:39:23 -0000	1.81.2.31
+++ admin/configuration.php	13 Apr 2010 14:00:04 -0000
@@ -35,7 +35,7 @@
 #
 
 define('USE_TRUSTED_POST_VARIABLES',1);
-$trusted_post_variables = array("gpg_key", "pgp_key");
+$trusted_post_variables = array("gpg_key", "pgp_key", "xpc_private_key_password", "xpc_private_key", "xpc_public_key");
 
 require "./auth.php";
 require $xcart_dir."/include/security.php";
@@ -76,6 +76,10 @@
 	require $xcart_dir."/include/safe_mode.php";
 }
 
+if ($option == 'XPayments_Connector') {
+	include $xcart_dir . '/modules/XPayments_Connector/xpc_admin.php';
+}
+
 if ($option == "User_Profiles") {
 	include "./user_profiles.php";
 }
Index: admin/payment_methods.php
===================================================================
RCS file: /usr/local/cvsroot/eshop/admin/payment_methods.php,v
retrieving revision 1.33.2.5
diff -u -r1.33.2.5 payment_methods.php
--- admin/payment_methods.php	5 Feb 2009 10:39:25 -0000	1.33.2.5
+++ admin/payment_methods.php	13 Apr 2010 14:00:04 -0000
@@ -44,6 +44,10 @@
 
 x_load('backoffice','tests');
 
+if (!empty($active_modules['XPayments_Connector'])) {
+	func_xpay_func_load();
+}
+
 $location[] = array(func_get_langvar_by_name("lbl_payment_methods"), "");
 
 if ($REQUEST_METHOD == "POST") {
@@ -89,7 +93,7 @@
 #
 # Obtain payment methods
 #
-$payment_methods = func_query("SELECT pm.*,cc.module_name,cc.processor,cc.type FROM $sql_tbl[payment_methods] AS pm LEFT JOIN $sql_tbl[ccprocessors] AS cc ON (pm.paymentid=cc.paymentid OR pm.paymentid<>cc.paymentid AND pm.processor_file=cc.processor) ORDER BY pm.orderby, pm.paymentid");
+$payment_methods = func_query("SELECT pm.*, cc.module_name, cc.processor, cc.type, cc.param01 FROM $sql_tbl[payment_methods] AS pm LEFT JOIN $sql_tbl[ccprocessors] AS cc ON (pm.paymentid=cc.paymentid OR pm.paymentid<>cc.paymentid AND pm.processor_file=cc.processor AND cc.processor != 'cc_xpc.php') ORDER BY pm.active DESC, pm.orderby, pm.paymentid");
 $payment_methods = test_payment_methods($payment_methods);
 
 #
@@ -154,6 +158,14 @@
 $cc_module_files = func_query("select * from $sql_tbl[ccprocessors] where paymentid='0' and processor<>'ps_paypal_pro.php' order by type,module_name");
 $sb_module_files = func_query("select * from $sql_tbl[ccprocessors] where type='C' and background='Y' order by module_name");
 
+if (!empty($active_modules['XPayments_Connector'])) {
+	foreach ($cc_module_files as $key => $value) {
+		if ($value['processor'] == 'cc_xpc.php') {
+			$cc_module_files[$key]['processor'] .= '_' . $value['param01'];
+		}
+	}
+}
+
 $smarty->assign("cc_modules",$cc_module_files);
 $smarty->assign("sb_modules",$sb_module_files);
 
Index: skin1/admin/main/configuration.tpl
===================================================================
RCS file: /usr/local/cvsroot/eshop/skin1/admin/main/configuration.tpl,v
retrieving revision 1.91.2.15
diff -u -r1.91.2.15 configuration.tpl
--- skin1/admin/main/configuration.tpl	12 Nov 2008 11:11:03 -0000	1.91.2.15
+++ skin1/admin/main/configuration.tpl	13 Apr 2010 14:00:11 -0000
@@ -71,6 +71,11 @@
 	<br />
 	<br />
 {/if}
+
+{if $option eq "XPayments_Connector"}
+  {include file="modules/XPayments_Connector/config_recommends.tpl"}
+{/if}
+
 <table cellpadding="3" cellspacing="1">
 
 {assign var="first_row" value=1}
@@ -313,6 +318,10 @@
 
 <a href="test_pgp.php">{$lng.lbl_test_data_encryption_link}</a>
 
+{elseif $option eq "XPayments_Connector"}
+
+  {include file="modules/XPayments_Connector/config_bottom.tpl"}
+
 {/if}
 
 <br />
