Index: include/func/func.order.php
=====================================================
--- include/func/func.order.php
+++ include/func/func.order.php
@@ -705,6 +705,15 @@
 		}
 
 		}
+  // Send order to Mailchimp if module enabled and mailchimp_analytics = Y
+                global $mailchimp_campaignid;
+                if (
+                  !empty($active_modules['Mailchimp_Subscription'])
+                  && $config['Mailchimp_Subscription']['mailchimp_analytics']
+                  && $mailchimp_campaignid
+                ) {
+                      func_mailchimp_adv_campaign_commission($orderid, $mailchimp_campaignid);
+                }
 
 		if (!empty($active_modules['XAffiliate'])) {
 			#
